Source/DafnyPipeline.Test/Boogie.cs

```diff
@@ -0,0 +1,219 @@
++     havoc k#0_0;
+-         assume y#0_0 < x#0;
++         assume k#0_0 < x#0;
+-         x#1_0 := LitInt(0);
++         a#1_0 := LitInt(0);
+-         assume !(exists eg$y#1: int :: eg$y#1 < x#0);
++         assume !(exists eg$k#1: int :: eg$k#1 < x#0);
+-         y#2_0 := LitInt(0);
++         b#2_0 := LitInt(0);
+-         x#3_0 := LitInt(2);
++         c#3_0 := LitInt(2);
+-         y#3_0 := LitInt(0);
++         d#3_0 := LitInt(0);
+-         x#4_0 := LitInt(2);
++         e#4_0 := LitInt(2);
+-         y#4_0 := LitInt(0);
++         f#4_0 := LitInt(0);
+-         x#5_0 := LitInt(2);
++         g#5_0 := LitInt(2);
+-         y#5_0 := LitInt(0);
++         h#5_0 := LitInt(0);
+-         x#7_0 := x#0 + 2;
++         i#7_0 := x#0 + 2;
+-         y#7_0 := LitInt(0);
++         j#7_0 := LitInt(0);
+";
+          Assert.Equal(expectedDiff, diff);
+        }
+
+        string GetDiff(string before, string after) {
+          var diff = InlineDiffBuilder.Diff(before, after);
+          var result = new StringBuilder();
+          foreach (var line in diff.Lines)
+          {
+            switch (line.Type)
+            {
+              case ChangeType.Inserted:
+                result.Append("+ ");
+                result.AppendLine(line.Text);
+                break;
+              case ChangeType.Deleted:
+                result.Append("- ");
+                result.AppendLine(line.Text);
+                break;
+            }
+
+          }
+
+          return result.ToString();
+        }
+
+        static Boogie() {
+          var testAssemblyPath = Assembly.GetAssembly(typeof(Boogie)).GetAssemblyLocation();
+          var parts = testAssemblyPath.Split(Path.DirectorySeparatorChar);
+          // This way of finding the repository root is not reliable, we should instead reference the DafnyPipeline assembly and run Dafny in the same process as the unit tests.
+          var sourceIndex = Array.FindIndex(parts, e => e == "Source");
+          dafnyDirectory = Path.Combine(Path.GetPathRoot(testAssemblyPath)!, Path.Combine(parts.Take(sourceIndex).ToArray()));
+          Console.WriteLine("dafnyDirectory: " + dafnyDirectory);
+          Console.WriteLine("DafnyDriverProjectFile: " + DafnyDriverProjectFile);
+        }
+
+        private static readonly string dafnyDirectory;
+
+        private static string DafnyDriverProjectFile => Path.Combine(dafnyDirectory, "Source", "DafnyDriver", "DafnyDriver.csproj");
+        private static string DefaultDafnyArgs => $"run --no-build --project {DafnyDriverProjectFile} -- -useBaseNameForFileName -countVerificationErrors:0 -compileVerbose:0 /errorTrace:0";
+        
+        string GetBoogie(string dafnyProgram, string optionalFileName = null) {
+          string fileName = optionalFileName ?? Path.GetTempFileName() + ".dfy";
+          var writer  = File.CreateText(fileName);
+          writer.Write(dafnyProgram);
+          writer.Flush();
+          var processStartInfo = new ProcessStartInfo
```

Comment:
FYI we can refactor this to use the generic `CLITestCase` class I'm adding as part of #680